packages/in_app_purchase/ios/Classes/InAppPurchasePlugin.m

```diff
@@ -40,19 +40,21 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
 }
 
 - (instancetype)initWithReceiptManager:(FIAPReceiptManager *)receiptManager {
-  self = [self init];
-  self.receiptManager = receiptManager;
+  self = [super init];
+  _receiptManager = receiptManager;
+  _requestHandlers = [NSMutableSet new];
+  _productsCache = [NSMutableDictionary new];
```

Comment:
curious, what is the benefit of initialize them in at initialization v.s. the lazy init that we used to have?

Reply:
In this particular case the lazy getter wasn't threadsafe.  Also this class assumes these collections are instantiated, but there's nothing stopping a subclass from overriding the getter to do something wonky that doesn't call super or set up the ivar (probably unlikely).  To me it looked like it was lazily instantiated because there was no designated initializer decoration on the init methods, so there was no one canonical spot where these collections could be relied on to be instantiated.

Assuming this object is queue confined (I don't know if that's actually enforced or just convention) it's fine to keep it lazily instantiated, and I can change it back if you prefer!

Reply:
Thanks for the explanation. What you have now is good! LGTM again.